File: mobile-portrait/android/app/src/main/java/com/miru/portrait/OutfitCommand.java
package com.miru.portrait;

final class OutfitCommand {
    private OutfitCommand() { }

    static String parse(String text, String current) {
        String input = text.replaceAll("\\s+", "");
        if (!(input.contains("換") || input.contains("穿") || input.contains("切換") || input.contains("改成"))
                || !(input.contains("衣") || input.contains("服") || input.contains("裝") || input.contains("造型"))) {
            return null;
        }
        if (input.contains("居家") || input.contains("家居") || input.contains("睡衣") || input.contains("家裡")) return "home";
        if (input.contains("外出") || input.contains("出門") || input.contains("外面")) return "outdoor";
        if (input.contains("一般") || input.contains("平常") || input.contains("原本") || input.contains("預設")
                || input.contains("普通")) return "normal";
        if ("normal".equals(current)) return "home";
        if ("home".equals(current)) return "outdoor";
        return "normal";
    }
}

File: assets/js/companion/outfit-command.js
(function(root, factory) {
  var api = factory();
  if (typeof module === 'object' && module.exports) module.exports = api;
  if (root) root.MiruOutfitCommand = api;
})(typeof globalThis !== 'undefined' ? globalThis : this, function() {
  'use strict';
  var outfits = ['normal','home','outdoor'];
  function parse(text,current) {
    var input = String(text || '').replace(/\s+/g,'');
    if (!/(換|穿|切換|改成|改穿)/.test(input) || !/(衣|服|裝|造型)/.test(input)) return null;
    if (/(居家|家居|睡衣|家裡)/.test(input)) return 'home';
    if (/(外出|出門|外面)/.test(input)) return 'outdoor';
    if (/(一般|平常|原本|預設|默认|普通)/.test(input)) return 'normal';
    var index = outfits.indexOf(current);
    return outfits[((index < 0 ? 0 : index) + 1) % outfits.length];
  }
  return {parse:parse};
});
